go-client/examples/query/main.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/deephaven/deephaven-core/go-client/client"
+	"github.com/deephaven/deephaven-core/go-client/examples/common"
+)
+
+// This example shows how to use the powerful query system,
+// which allows you to perform an arbitrary number of table operations in a single request.
+// It can even create new tables as part of a query,
+// and the interface is more convenient than doing each of the table operations serially.
+func main() {
+	// If you don't have any specific requirements, context.Background() is a good default.
+	ctx := context.Background()
+
+	cl, err := client.NewClient(ctx, "localhost", "10000", "python")
+	if err != nil {
+		fmt.Println("error when connecting to localhost port 10000:", err.Error())
+		return
+	}
+	defer cl.Close()
+
+	// Let's get some example data to play with.
+	sampleRecord := common.GetExampleRecord()
+	defer sampleRecord.Release()
+
+	fmt.Println("Data Before:")
+	fmt.Println(sampleRecord)
+
+	// We do need to upload the table data normally, without using a query.
+	baseTable, err := cl.ImportTable(ctx, sampleRecord)
+	if err != nil {
+		fmt.Println("error when uploading table:", err.Error())
+		return
+	}
+	defer baseTable.Release(ctx)
+
+	// And now the fun begins.
```

Comment:
A user going through this example is going to be lost.  Every good example is a story.  You want the story to be very easy to follow.  If you read the comment on main and then jump in here, the reader will not have a storyline to follow.  The comments should help the story be clear and move forward.